reconcile/test/test_utils_terraform_client.py

```diff
@@ -251,3 +251,258 @@ def test__resource_diff_changed_fields(tf):
     )
 
     assert changed == set()
+
+
+def test__get_pending_rds_modification(aws_api, tf, caplog, capsys):
+    with patch.object(aws_api, "describe_rds_db_instance"):
```

Comment:
With `patch` it's generally always good to use autospec unless there's a good reason not to. There's more on this in our style guide: https://service.pages.redhat.com/dev-guidelines/docs/appsre/python-testing-guide/#mock-as-little-as-possible

The short version is that setting autospec will ensure that the mock, returned from `patch`, cannot be called with the wrong number of arguments, or in the case of classes, will ensure that there isn't access to members/methods that don't exist.

Reply:
@steveteahan, very good pointers! I will refactor the code accordingly.

Reply:
Done. Thank you!